### PR TITLE
refactor: validate request parameters

### PR DESCRIPTION
**Pull request by codingben**
Add validation of the request parameters in API endpoints, and do slightly refactor of the endpoints to improve code readability.

Issue #13.

### CODE REVIEW

**Comment by codingben** [comment]
Converted to draft because I need to check failures of the unit tests. Until then, it can be reviewed if you wish to do.

**Review by fran2810 — COMMENTED** [review]
fix error issue because now we cant parse any error only k8s errors.

File: src/routes/v1/secret.go [hunk 1]
```diff
@@ -43,57 +43,75 @@ func secretHandler(handler func(controller controllers.SecretController, c *gin.
...
-	return secretHandler(func(controller controllers.SecretController, c *gin.Context) (interface{}, error) {
-		namespaceName := c.Param("namespaceName")
-		secretName := c.Param("secretName")
-
+	return func(c *gin.Context) {
 		var request types.PatchSecretRequest
-		if err := c.ShouldBindJSON(&request); err != nil {
-			c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error()})
-			return nil, err
+		if err := c.BindJSON(&request); err != nil {
```
Comment (fran2810) [inline src/routes/v1/secret.go:93]: there`s no validation to URI here.  and the route should accept namespace name and secret name. you should divide to two struct and unify them after two validations function

**Review by fran2810 — APPROVED** [review]
/lgtm

**Comment by codingben** [comment]
@mzeevi Can you please review and approve, if all good?

**Review by mzeevi — APPROVED** [review]

**Pull request merged**